pontoon/terminology/formats/tbx.py

```diff
@@ -0,0 +1,192 @@
+"""
+This module tries to implement subset of The TBX Basic format.
+
+First version of the implementation allows to import terminology from The Microsoft Language Portal.
+However, there's a lot corner-cases that aren't handled and they will be implemented
+in future versions.
+As the name suggest, TBX is the xml compliant format, that can be parsed by any xml library.
+We try to wrap all xml structures in python classes to make the code more readable.
+"""
+from collections import namedtuple
+
+from defusedxml.minidom import parseString
+
+
+class MissingSourceTerm(Exception):
+    """
+    When Term doesn't contain Language translation for en-GB or en-US.
+    """
+
+
+class XMLObject(object):
+    """
+    All tbx objects will share the same constructor that will receive an instance of a xml object.
+    """
+    def __init__(self, xml_node):
+        self.xml_node = xml_node
+
+    @property
+    def description(self):
+        """
+        Most of the object in tbx file have a description.
+        """
+        for child in self.xml_node.childNodes:
+            if child.nodeType == child.TEXT_NODE:
+                continue
+
+            if child.tagName == 'descrip':
+                try:
+                    return child.childNodes[0].data
+                except IndexError:
+                    return ''
+
+            elif child.tagName == 'descripGrp':
+                return child.childNodes[0].childNodes[0].data
+
+
+class Translation(XMLObject):
+    """
+    Translation of term to a given language. Also contains information about the part of speech.
+    """
+    @property
+    def text(self):
+        try:
+            return self.xml_node.getElementsByTagName('term')[0].childNodes[0].data
+        except IndexError:
+            return ''
+
+    @property
+    def notes(self):
+        notes_ = {}
+        for note in self.xml_node.getElementsByTagName('termNote'):
+            note_type = str(note.getAttribute('type'))
+            note_text = str(note.childNodes[0].data)
```

Comment:
I think I'd rather be safe pretty much everywhere you're using the XML structure. It's hard to rely on those files to be all perfectly formatted, and it would be better to have failsafe cases everywhere.